netbox_netprod_importer/devices_list.py

```diff
@@ -28,3 +31,57 @@ def parse_devices_yaml_def(devices_yaml, creds=None):
                     "Cannot connect to device %s: %s", hostname, e
                 )
     return devices
+
+
+def parse_filter_yaml_def(filter_yaml, creds=None):
+    netbox_api = NetboxAPI(**get_config()["netbox"])
+    devices = {}
+    with open(filter_yaml) as filter_yaml_str:
+        q_list = []
+        yml = yaml.safe_load(filter_yaml_str)
+        for fil, props in tqdm(yml["filter"].items()):
+            if props is not None:
+                if type(props) is list:
+                    q_list.append(fil + "=" +
+                                  ("&" + fil + "=").join(str(x) for x in props)
+                                  )
+                else:
+                    q_list.append(fil + "=" + str(props))
+        logger.info("Resultant filter %s", "&".join(q_list))
+        platforms_js = netbox_api.get("dcim/platforms")
+        platforms = {}
+        platforms_args = {}
+        for platform in platforms_js["results"]:
+            if platform["napalm_driver"]:
+                platforms[platform["id"]] = platform["napalm_driver"]
+                platforms_args[platform["id"]] = platform["napalm_args"]
+        if len(platforms) == 0:
+            logger.error("Not for one platform napalm_driver is not defined")
+            sys.exit(4)
+
+        devlist = netbox_api.get("dcim/devices/?" + "&".join(q_list))
+        for device in devlist["results"]:
+            if not device["platform"]:
```

Comment:
You can put the 2 if on the same line:
```
if not device.get("platform") or not platforms.get(device["platform"]["id"]):
    continue
```

Reply:
There will be an exception if device ["platform"] is Null

Reply:
No, because if the first condition is `True` (and `not device.get("platform")` is True if it's null), python will not continue if it's a `or`, as it's not necessarily. For example:
```
>>> if not None or not None.copy():
...   print("foo")
...
foo
>>> if None or not None.copy():
...   print("foo")
...
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'NoneType' object has no attribute 'copy'
```

`None.copy()` is never reached in the first block, because `not None` is True, so there is no need to continue.

Reply:
I did not know this nuance of the interpreter behavior. Done.